aes/src/ni/utils.rs

```diff
@@ -1,5 +1,8 @@
 //! Utility functions
 
+// TODO(tarcieri): check performance impact / generated assembly changes
+#![allow(clippy::needless_range_loop)]
```

Comment:
Strangely clippy's suggested refactoring seemed to negatively impact performance?(!)

Reply:
Yeah, I've seen such behavior as well. Iterator based code probably makes it harder for LLVM to unroll the loops. Also subjectively, I think for such loops range loops result in a better, easier to understand code.